src/main/java/net/minecraftforge/client/model/animation/Animation.java

```diff
@@ -0,0 +1,186 @@
+package net.minecraftforge.client.model.animation;
+
+import net.minecraft.util.ResourceLocation;
+import net.minecraft.world.World;
+import net.minecraftforge.client.model.IModelState;
+import net.minecraftforge.client.model.animation.ModelBlockAnimation.MBJointInfo;
+import net.minecraftforge.common.property.IUnlistedProperty;
+
+import com.google.common.base.Optional;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableTable;
+
+public class Animation
+{
+    public static final IUnlistedProperty<IModelState> AnimationProperty = new IUnlistedProperty<IModelState>()
+    {
+        public String getName() { return "forge_animation"; }
+        public boolean isValid(IModelState state) { return true; }
+        public Class<IModelState> getType() { return IModelState.class; }
+        public String valueToString(IModelState state) { return state.toString(); }
+    };
+
+    public static float getWorldTime(World world)
+    {
+        return getWorldTime(world, 0);
+    }
+
+    public static float getWorldTime(World world, float tickProgress)
+    {
+        return (world.getTotalWorldTime() + tickProgress) / 20;
+    }
+
+    private static final AnimationStateMachine missing = new AnimationStateMachine(
```

Comment:
I simply followed the missing texture/missing model, ehm, model. It's useful to have a default resource when the loading fails, and let the rest of the code still work, without indeed special-casing the missing resource everywhere.
Might not be the best strategy for the animation state machine though, since the logic depends on the set of available states right now. I'm happy to change this to something more appropriate - this is why it's a PR :P